api/src/backend/api/tests/test_views.py

```diff
@@ -5041,3 +5041,277 @@ def test_integrations_filters_invalid(self, authenticated_client, filter_name):
+            content_type=API_JSON_CONTENT_TYPE,
+        )
+        assert response.status_code == status.HTTP_201_CREATED
+        data = response.json()["data"]
+        assert data["attributes"]["name"] == "Test Config"
+        assert data["attributes"]["model"] == "gpt-4o"
+        assert data["attributes"]["temperature"] == 0.7
+        assert data["attributes"]["max_tokens"] == 4000
+        assert data["attributes"]["business_context"] == "Test business context"
+        assert data["attributes"]["is_active"] is True
+        assert data["attributes"]["api_key"] == "*" * len(
+            valid_config_payload["data"]["attributes"]["api_key"]
+        )
+
+    def test_lighthouse_config_create_duplicate(
+        self, authenticated_client, valid_config_payload
+    ):
+        # Create first config
+        response = authenticated_client.post(
+            "/api/v1/lighthouse-config/",
+            data=valid_config_payload,
+            content_type=API_JSON_CONTENT_TYPE,
+        )
+        assert response.status_code == status.HTTP_201_CREATED
+
+        # Try to create second config for same tenant
+        response = authenticated_client.post(
+            "/api/v1/lighthouse-config/",
+            data=valid_config_payload,
+            content_type=API_JSON_CONTENT_TYPE,
+        )
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert (
+            "AI configuration already exists for this tenant"
+            in response.json()["errors"][0]["detail"]
+        )
+
+    def test_lighthouse_config_create_invalid(
+        self, authenticated_client, invalid_config_payload
+    ):
+        response = authenticated_client.post(
+            "/api/v1/lighthouse-config/",
+            data=invalid_config_payload,
+            content_type=API_JSON_CONTENT_TYPE,
+        )
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        errors = response.json()["errors"]
+        assert any("name" in error["source"]["pointer"] for error in errors)
+        assert any("api_key" in error["source"]["pointer"] for error in errors)
+
+    def test_lighthouse_config_retrieve(
+        self, authenticated_client, valid_config_payload
+    ):
+        # Create config
+        response = authenticated_client.post(
+            "/api/v1/lighthouse-config/",
+            data=valid_config_payload,
+            content_type=API_JSON_CONTENT_TYPE,
+        )
+        assert response.status_code == status.HTTP_201_CREATED
+        config_id = response.json()["data"]["id"]
+
+        # Retrieve config
+        response = authenticated_client.get(f"/api/v1/lighthouse-config/{config_id}/")
+        assert response.status_code == status.HTTP_200_OK
+        data = response.json()["data"]
+        assert data["attributes"]["name"] == "Test Config"
+        assert data["attributes"]["api_key"] == "*" * len(
+            valid_config_payload["data"]["attributes"]["api_key"]
+        )
```

Comment:
This is not a unit test if you depend on a POST for a retrieval. Use pytest fixtures instead so you already have items to work with in the database when the test begins.

Reply:
Apply this principle on every other test, please.

Reply:
I've updated the tests, please check. Thanks!